lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/SettingsService.java

```diff
@@ -64,4 +80,63 @@ public int getRequestDelay() {
         return DEFAULT_REQUEST_DELAY;
     }
 
+    /**
+     * populate all variables for all available workspace folders
+     *
+     * @param workspaceFolders workspace folders
+     */
+    public void populateAllVariables(Collection<LibertyWorkspace> workspaceFolders) {
+        variables = new HashMap<>();
+        for (LibertyWorkspace workspace : workspaceFolders) {
+            populateVariablesForWorkspace(workspace);
+        }
+    }
+
+    /**
+     * read all variables from workspace directories
+     *
+     * @param workspace workspace
+     */
+    public void populateVariablesForWorkspace(LibertyWorkspace workspace) {
+        Properties variablesForWorkspace = new Properties();
+        Path pluginConfigFilePath = findFileInWorkspace(workspace, Paths.get("liberty-plugin-config.xml"));
+        if (pluginConfigFilePath != null) {
+            File installDirectory = LibertyUtils.getFileFromLibertyPluginXml(pluginConfigFilePath, "installDirectory");
+            File serverDirectory = LibertyUtils.getFileFromLibertyPluginXml(pluginConfigFilePath, "serverDirectory");
+            File userDirectory = LibertyUtils.getFileFromLibertyPluginXml(pluginConfigFilePath, "userDirectory");
+            if (serverDirectory != null && installDirectory != null && userDirectory != null) {
+                try {
+                    ServerConfigDocument serverConfigDocument = new ServerConfigDocument(
+                            new CommonLogger(LOGGER), null, installDirectory, userDirectory, serverDirectory);
+                    variablesForWorkspace.putAll(serverConfigDocument.getDefaultProperties());
+                    variablesForWorkspace.putAll(serverConfigDocument.getProperties());
+                } catch (Exception e) {
+                    LOGGER.warning("Variable resolution is not available because the necessary directory locations were not found in the liberty-plugin-config.xml file.");
+                    LOGGER.info("Exception received: " + e.getMessage());
+                }
+            }
+        } else {
+            LOGGER.warning("Could not find liberty-plugin-config.xml in workspace URI" + workspace.getWorkspaceString() + ". Variable processing cannot be performed");
```

Comment:
Let's change it to be consistent and say "Variable resolution...." and add a period at the end of the message. Also need a space between `URI` and the `"` so the actual URI is not smashed together.

Reply:
Done